File: frontend/src/api/project.ts
import api from './index'

export interface ProjectItem {
  id?: number
  projectCode: string
  projectName: string
  projectType: string
  securityLevel: string
  status: string
  ownerUserId: string
  applicableStandards: string
  startDate: string
  endDate: string
  description: string
}

export function getProjects(params: { pageNo?: number; pageSize?: number; keyword?: string; status?: string }) {
  return api.get('/projects', { params })
}

export function getProject(id: number) {
  return api.get(`/projects/${id}`)
}

export function createProject(data: ProjectItem) {
  return api.post('/projects', data)
}

export function updateProject(id: number, data: ProjectItem) {
  return api.put(`/projects/${id}`, data)
}

File: backend/src/main/java/com/military/doc/modules/project/entity/Project.java
package com.military.doc.modules.project.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("project")
public class Project {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String projectCode;
    private String projectName;
    private String projectType;
    private String securityLevel;
    private String status;
    private String ownerUserId;
    private String applicableStandards;
    private LocalDate startDate;
    private LocalDate endDate;
    private String description;

    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}